source/module_elecstate/module_charge/charge_mixing.cpp

```diff
@@ -82,54 +92,55 @@ void Charge_Mixing::set_mixing(const std::string& mixing_mode_in,
     return;
 }
 
-void Charge_Mixing::set_rhopw(ModulePW::PW_Basis* rhopw_in)
+void Charge_Mixing::set_rhopw(ModulePW::PW_Basis* rhopw_in, ModulePW::PW_Basis* rhodpw_in)
 {
     this->rhopw = rhopw_in;
+    this->rhodpw = rhodpw_in;
 }
 
-void Charge_Mixing::need_auto_set()
-{
-    this->autoset = true;
-}
-
-void Charge_Mixing::auto_set(const double& bandgap_in, const UnitCell& ucell_)
-{
-    // auto set parameters once
-    if (!this->autoset)
-    {
-        return;
-    }
-    else
-    {
-        this->autoset = false;
-    }
-    GlobalV::ofs_running << "--------------AUTO-SET---------------" << std::endl;
-    // 0.8 for nspin=1 and 0.4 for others
-    if (GlobalV::NSPIN == 1)
-    {
-        this->mixing->mixing_beta = this->mixing_beta = 0.8;
-        GlobalV::ofs_running << "      Autoset mixing_beta to " << this->mixing_beta << std::endl;
-    }
-    else
-    {
-        this->mixing->mixing_beta = this->mixing_beta = 0.4;
-        GlobalV::MIXING_BETA_MAG = 4 * this->mixing_beta;
-        GlobalV::ofs_running << "      Autoset mixing_beta to " << this->mixing_beta << std::endl;
-        GlobalV::ofs_running << "      Autoset mixing_beta_mag to " << GlobalV::MIXING_BETA_MAG << std::endl;
-    }
-    GlobalV::MIXING_BETA = mixing_beta;
-
-    // auto set kerker mixing_gg0 = 1.0 as default
-    this->mixing_gg0 = 1.0;
-    GlobalV::ofs_running << "      Autoset mixing_gg0 to " << this->mixing_gg0 << std::endl;
-    if (GlobalV::NSPIN == 1)
-    {
-        GlobalV::MIXING_GG0_MAG = 0.0;
-        GlobalV::ofs_running << "      Autoset mixing_gg0_mag to " << GlobalV::MIXING_GG0_MAG << std::endl;
-    }
-
-    GlobalV::ofs_running << "-------------------------------------" << std::endl;
-}
+// void Charge_Mixing::need_auto_set()
```

Comment:
I agree to delete need_auto_set() and auto_set(), which will never be used in the future.